### PR TITLE
Add lint-staged and husky for dev efficiency

### PR DESCRIPTION
**Pull request by edellucien**

### CODE REVIEW

File: .huskyrc.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "hooks": {
+    "post-checkout": "yarn install --frozen-lockfile",
```
Comment (friedjoff) [inline .huskyrc.json]: Maybe use HUSKY_GIT_PARAMS to check if git is checking out a branch (see https://youtu.be/ikn_dBSski8?t=780 for details)?

Comment (edellucien) [reply .huskyrc.json]: Better keep consistent the way we use those dep between our project. I include it in the package.json then :+1:

File: .lintstagedrc.json [hunk 2]
```diff
@@ -1,9 +0,0 @@
-{
-  "src/**/*.js": [
-    "eslint --fix",
-    "prettier --write",
-    "git add",
-    "yarn test --bail --findRelatedTests"
-  ],
-  "src/**/*.{css,scss}": ["stylelint --fix", "git add"]
```
Comment (friedjoff) [inline .lintstagedrc.json]: Did you remove the stylelint on purpose?

File: .huskyrc.json (hunk 1 above)
Comment (friedjoff) [reply .huskyrc.json]: Good idea!

**Comment by oterral** [comment]
@edellucien  can you fix the conflict please

**Pull request merged**